tests/integration/components/moderators/component-test.ts

```diff
@@ -142,4 +143,79 @@ module('Integration | Component | moderators', hooks => {
         assert.dom('[data-test-permission-group]').exists({ count: 1 });
         assert.dom('[data-test-delete-moderator-button]').doesNotExist();
     });
+
+    test('can add moderator by searching for user', async function(
+        this: ModeratorsTestContext, assert,
+    ) {
+        const currentUser = server.create('user', { id: 'sprout' });
+        const currentUserModel = await this.store.findRecord('user', 'sprout');
+        this.owner.lookup('service:current-user').setProperties({
+            user: currentUserModel, currentUserId: currentUserModel.id,
+        });
+        const admin = server.create('moderator', {
+            id: currentUserModel.id,
+            user: currentUser,
+        }, 'asAdmin');
+        this.provider.update({
+            moderators: [admin],
+        });
+        const user = server.create('user', { fullName: 'Hwang Yeji' });
+        await render(
+            hbs`<Moderators::Manager
+                    @provider={{this.providerModel}}
+                    as |moderatorManager|
+                >
+                    <Moderators::List
+                        @manager={{moderatorManager}}
+                    />
+                </Moderators::Manager>`,
+        );
+        assert.dom('[data-test-add-moderator-button]').exists();
+        assert.dom('[data-test-moderator-link]').exists({ count: 1 });
+        await click('[data-test-add-moderator-button]');
+        assert.dom('#osf-dialog-heading').hasText('Add a moderator');
+        await selectSearch('[data-test-select-user]', 'Yeji');
+        await selectChoose('[data-test-select-user]', 'Hwang Yeji');
+        await selectChoose('[data-test-select-permission]', 'Moderator');
+        await click('[data-test-confirm-add-moderator-button]');
+        assert.dom('[data-test-moderator-link]').exists({ count: 2 });
+        assert.dom(`[data-test-moderator-row="${user.id}"]`).exists();
+    });
+
+    test('can add moderator by email invite', async function(
+        this: ModeratorsTestContext, assert,
+    ) {
+        const currentUser = server.create('user', { id: 'sprout' });
+        const currentUserModel = await this.store.findRecord('user', 'sprout');
+        this.owner.lookup('service:current-user').setProperties({
+            user: currentUserModel, currentUserId: currentUserModel.id,
+        });
```

Comment:
Doesn't using `loggedIn` trait work here?
```suggestion
        const currentUser = server.create('user', { id: 'sprout' }, 'loggedIn');
```

Reply:
The `loggedIn` trait populates the `current_user` field for the root `/v2` endpoint. When the app boots, it always make a request to the root `/v2` endpoint, therefore the `currentUser` service will be instantiated with the respective userId. However because this is an integration test, the that root `/v2` request is never made. Creating a user with `loggedIn` trait would not instantiate the `currentUser` service.

Reply:
Gotcha! `loggedIn` is better suited for acceptance tests.